### PR TITLE
multiplier

### PR DESCRIPTION
**Pull request by MarkusRabe**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ an bitvector input, `'x'` with 3 bits becomes inputs `'x[0]', 'x[1]', 'x[3]'`
 
 # Installation
 
-If you just need to use `aiger_bv`, you can just run:
+If you just need to use `aigerbv`, you can just run:
```
Comment (mvcisback) [inline README.md]: Was this an intentional change? The current module is `aiger_bv`

File: aiger_bv/expr.py [hunk 2]
```diff
@@ -33,6 +33,8 @@ def __getitem__(self, idx: Union[int, slice]):
         def _indexer(idx):
             return cmn.index_gate(self.size, idx, self.output, cmn._fresh())
 
+        if isinstance(idx, int) and idx < 0:
+            idx = idx % self.size
```
Comment (mvcisback) [inline aiger_bv/expr.py:40]: Could you explain the motivation/use case here? Shouldn't we throw an error if `idx >= self.size`.

Comment (mvcisback) [reply aiger_bv/expr.py:40]: Ah I see, this is for negative indexing.  Why check if `idx < 0`?

Perhaps instead:

```python
if isinstance(idx, int):
    assert -self.size <= idx < self.size
    idx %= self.size    # Allow for indexing from end of bit-vector.
```

**Review by mvcisback — CHANGES_REQUESTED** [review]
Looks good overall!

File: README.md (hunk 1 above)
Comment (MarkusRabe) [reply README.md]: Further down in the README 'import aigerbv' is used. I quickly tried both and on my machine only  the variant aigerbv worked.

Comment (MarkusRabe) [reply README.md]: Import aiger_bv only works when you happen to be in the root directory of the package or repo, when it matches the directory name.

File: aiger_bv/expr.py (hunk 2 above)
Comment (MarkusRabe) [reply aiger_bv/expr.py:40]: Good point. idx >= self.size should throw an error.

I'll implement it with a raise ValueError instead, if that's okay.

File: README.md (hunk 1 above)
Comment (mvcisback) [reply README.md]: @MarkusRabe I have a strong suspicion you have an old version of the package installed.

I would recommend doing:
`pip show py-aiger-bv`

or 

`pip freeze | grep py-aiger-bv`

and making sure you have `2.0.0` installed.

File: aiger_bv/expr.py (hunk 2 above)
Comment (mvcisback) [reply aiger_bv/expr.py:40]: Sounds good!

File: README.md (hunk 1 above)
Comment (MarkusRabe) [reply README.md]: Yes, turned out I had an old version. I'll update the PR shortly. Made the naming consistently aiger_bv throughout the README.

**Review by mvcisback — APPROVED** [review]
Looks good.

Runs fine locally. Need to update the multiply deadline, but I can do that after merging.

Will also try to factor the multiply gate through `binary_gate` to get things like `x * 2` to work.

**Comment by MarkusRabe** [comment]
I was thinking about implementing the multiplier as a gadget and then use binary_gate, but it looked pretty complicated to implement things on this level. It looks almost like the gadgets are a separate layer of the library implementation. Is there a way to simplify this? I imagine users of the library want to implement their own operations or variants of existing operations and this should be as simple as possible.

**Comment by mvcisback** [comment]
@MarkusRabe 

You know, I've been thinking about this as well. I think actually more and more we should try to flip this around. Have the "gate" version be defined using the `expr` DSL.

In fact, I've considered deprecating a lot of the `*_gate` functions.

For example, you would replace:
```python
gate = adder_gate(4, 'x', 'y', 'z')
```

with:

```python
x, y = atom(4, 'x'), atom(4, 'y')
gate = (x + y).with_output('z').aigbv
```

**Pull request merged**